lib/welcome/pages/register_page.dart

```diff
@@ -0,0 +1,205 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_app/util/email_field.dart';
+import 'package:flutter_app/util/loading_button.dart';
+import 'package:flutter_app/welcome/pages/login_page.dart';
+import 'package:flutter_app/util/password_field.dart';
+import 'package:flutter_app/util/supabase.dart';
+import 'package:progress_state_button/progress_button.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+class RegisterPage extends StatefulWidget {
+  const RegisterPage({super.key});
+
+  @override
+  State<RegisterPage> createState() => _RegisterPageState();
+}
+
+class _RegisterPageState extends State<RegisterPage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Register"),
+      ),
+      body: const Center(
+        child: SingleChildScrollView(
+          child: Padding(
+            padding: EdgeInsets.symmetric(horizontal: 40, vertical: 20),
+            child: RegisterForm(),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class RegisterForm extends StatefulWidget {
+  const RegisterForm({super.key});
+
+  @override
+  State<RegisterForm> createState() => _RegisterFormState();
+}
+
+class _RegisterFormState extends State<RegisterForm> {
+  final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
+  final emailController = TextEditingController();
+  final usernameController = TextEditingController();
+  final passwordController = TextEditingController();
+  final passwordConfirmationController = TextEditingController();
+  ButtonState _state = ButtonState.idle;
+
+  void onSubmit() async {
+    if (_formKey.currentState!.validate()) {
+      late final AuthResponse res;
+      try {
+        setState(() {
+          _state = ButtonState.loading;
+        });
+        res = await supabaseClient.auth.signUp(
+            password: passwordController.text, email: emailController.text);
+      } on AuthException {
+        fail();
+        showSnackBar("Something went wrong.");
```

Comment:
Do we want to tell the user if the mail already exists?

Reply:
we'll have to ask the AG about that, not sure... Added a TODO comment for now